### DISCUSSION TITLE
Clarification of "auto escape for non Melody patterns"

### DISCUSSION THREAD

**Discussion by nlc** [yoav-lavi/melody#39] [Q&A] [answered]
First off, kudos on the effort to ease the suffering of people that have to use regular expressions! I'm sure this project will be much appreciated.

I noticed that certain literals were being passed through unescaped, e.g. `some of ".";` results in `/.+/` instead of `/\.+/`. I looked for prior mention before creating an issue, and saw that one of the to-do items was "auto escape for non Melody patterns." I was unclear as to whether that referred to escaping literals or not. Should I create an issue, or is this already under development?

**Comment by yoav-lavi** [comment, marked as answer]
Thank you!

Yes, that's exactly the idea, automatically escaping the content of literals as needed (and perhaps providing a raw literal that isn't escaped)

It's not in development yet but will be

**Reply by nlc** [reply]
Excellent, thank you. Just wanted to make sure.

**Comment by yoav-lavi** [comment]
For the time being `some of "\."` will work

**Comment by yoav-lavi** [comment]
@nlc version `v0.6.0` has auto escaping 😃

https://github.com/yoav-lavi/melody/blob/main/CHANGELOG.md#v060---2022-02-23
